src/IROperator.cpp

```diff
@@ -2593,10 +2593,26 @@ Expr mod_round_to_zero(Expr x, Expr y) {
                       Call::PureIntrinsic);
 }
 
+namespace {
+
+std::atomic<int> random_float_counter = 0;
```

Comment:
Now that it's a global, this can just be one counter, and the even/odd nonsense below can be removed.